### PR TITLE
fix: Change panicking on unknown module to error print and support module listing flag

### LINKED ISSUE #190: `starship module` with unknown name panics
#### Current Behavior
Running `starship module sdf` panics with an "Unknown module" error.

#### Expected Behavior
It should print a user-friendly error.

#### Additional context/Screenshots
```
➜ starship module sdf
thread 'main' panicked at 'Unknown module: sdf', src/modules/mod.rs:38:14
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

#### Environment
- Starship version: starship 0.10.1
- Shell type: fish
- Shell version: fish, version 3.0.2
- Shell plugin manager: fisher
- Terminal emulator: Terminal.app
- Operating system: macOS 10.14.6 (18G87)

#### Starship Configuration
No configuration.

### PR DESCRIPTION
**Pull request by iamsauravsharma**
#### Description
Instead of panicking out over unknown module supplied to `starship module` subcommand this PR print outs standard error over terminal. This PR also adds out new flag for module subcommand --list i.e `starship module --list` which list out all of supported modules.

#### Motivation and Context
Closes #190 

#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?
- [ ] I have tested using **MacOS**
- [X] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: src/module.rs [hunk 1]
```diff
@@ -4,6 +4,25 @@ use ansi_term::Style;
...
+    "git_status",
+    "golang",
+    "jobs",
+    "line_break",
+    "nodejs",
+    "package",
+    "python",
+    "ruby",
+    "rust",
+    "username",
```
Comment (sirMerr) [inline src/module.rs]: Would it make sense to get this dynamically instead?

Comment (matchai) [reply src/module.rs]: As it is, there is no way to get this dynamically.
It may be a good idea to make this an enum in a future refactor. 👍

**Comment by sirMerr** [comment]
Thanks for tackling this :D! Could you add an integration test?

**Comment by matchai** [comment]
Looks great! Thank you for adding this feature, Saurav. 😄

**Pull request merged**